src/khepri.erl

```diff
@@ -174,6 +175,11 @@
 
                            unwrap_result/1]}).
 
+-deprecated({wait_for_async_ret, 1, "use khepri:handle_async_ret/2 and "
+                                    "`receive' Ra events instead"}).
+-deprecated({wait_for_async_ret, 2, "use khepri:handle_async_ret/2 and "
+                                    "`receive' Ra events instead"}).
+
```

Comment:
Given we didn't cut version 1.0.0 yet, we don't really need to follow a deprecation period. I think we can just get rid of these functions.